Feature/update docs after refactor

bkarambe opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes # BLUI-3670

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Updated Readme for icons and packages
- Added set of finite svgs inside react demo

<!-- Include screenshots if they will help illustrate the changes in this PR -->
#### Screenshots:
<img width="1792" alt="Screenshot 2022-12-19 at 7 57 12 PM" src="https://user-images.githubusercontent.com/25982779/208448080-01ddf310-2a10-4c95-9fef-531de25e8e49.png">

<!-- Instruction for PR reviewers, if more complicated than a simple yarn start -->
#### To Test:
- yarn
- yarn build
- yarn start

jeffvg commented:
after build should I have an icon-font folder?

- To test
- copy files inside the node_modules/@brightlayer-ui/icons/icon-font to node_modules/@brightlayer-ui/icons/icon-font

<img width="349" alt="image" src="https://user-images.githubusercontent.com/37150565/207431856-5dc5ae21-faa4-40c0-acad-72ca4ca96837.png">

joebochill commented:
> copy files inside the node_modules/@brightlayer-ui/icons/icon-font to node_modules/@brightlayer-ui/icons/icon-font

There should be absolutely NO manual copying of files into node modules after this refactoring. That's the entire reason we did the refactor. Do no do this.

jeffvg commented:
> > copy files inside the node_modules/@brightlayer-ui/icons/icon-font to node_modules/@brightlayer-ui/icons/icon-font
> 
> There should be absolutely NO manual copying of files into node modules after this refactoring. That's the entire reason we did the refactor. Do no do this.

ok no problem, but should there be a icon-font folder within node_modules?

After the build I ran ```yarn start``` 

![image](https://user-images.githubusercontent.com/37150565/207435069-dbd6f84c-7f39-45b8-a9b1-0e24cbcefdaf.png)

joebochill commented:
> after build should I have an icon-font folder?

After running the top-level build command (which builds all of the various icon repositories), you should have:

- a /packages/icon-font folder with font files in it (these are updated and replaced during the build command)
- a /packages/mui/dist folder (this is where the final result of the MUI icons are put — this is where the package gets published from. There is also an intermediate build folder /packages/mui/tsx as well)
- /packages/rn-vector/Font and /packages/rn-vector/GlyphMaps folders (THIS PART OF THE BUILD IS BROKEN AND NEEDS TO BE FIXED...it's trying to copy files from the wrong location. We also need to create those files if they don't exist or it will copy the file to a file instead of a folder)
- /packages/svg folder with raw svg files in it (these are updated if there are any changes or additions in the design folder)

joebochill commented on .circleci/config.yml:
```diff
@@ -55,7 +55,7 @@ jobs:
                       echo "//registry.yarnpkg.com/:_authToken=$NPM_TOKEN" >> ~/.npmrc
             - run:
                   name: Publish @brightlayer-ui/icons
-                  command: yarn publish:package -b $CIRCLE_BRANCH
```
missing &&

joebochill commented:
There is a subtle change for how the icon font repo is published now. Previously, the package.json for that was at the root of the repository and we published from the root and included the subfolder called iconfont (or icon-font, can't remember). So importing the package, you had to import the font file from the subfolder (import 'CSSFILE.css' from '@brightlayer-ui/icons/iconfont').

THAT IS NO LONGER THE CASE.

We moved the icon-font folder into a dedicated package folder under /packages and it gets published from that folder directly (no more subfolder). So the new usage is: import 'CSSFILE.css' from '@brightlayer-ui/icons'

joebochill commented on VALIDATION.md:
```diff
@@ -3,14 +3,21 @@
 #### React Native Vector Icons
-1. Validate a beta package integrates into the showcase demo.
 
+1. Validate a beta package integrates into the react demo.
```
You can't use RN vector icons in a react project. Change this to say something like a new RN project from the CLI instead.

joebochill commented on demos/react-demo/src/pages/svg.tsx:
```diff
@@ -5,9 +5,137 @@ import { Theme } from '@mui/material/styles';
 import Breaker from '@brightlayer-ui/icons-svg/breaker.svg';
 
-const meta = require('@brightlayer-ui/icons-mui/index.json');
+const svgIcons = [
```
Scale this array down to include only the data that is necessary to render the icons on the screen (there is a bunch of unnecessary data in here that just makes the file harder to read). Or pull the data from the actual meta file (using something like array.filter to filter by the names we want).

joebochill commented on packages/icon-font/CHANGELOG.md:
```diff
@@ -1,5 +1,11 @@
+
+### Changed
+
+-   Changed import path of icon font from `icons/iconfont` to `icons`.
```
Show a realistic code snippet example for the import to be very explicit

joebochill commented on packages/mui/PUBLISHING.md:
````diff
@@ -14,7 +14,7 @@ If you need to publish a package manually, you can run the following commands fr
 ```sh
 yarn build
-cd mui && yarn build
+cd mui
````
```suggestion
cd mui/dist
```

joebochill commented on packages/mui/PUBLISHING.md:
````diff
@@ -24,7 +24,7 @@ For this command to work, you must have an NPM token configured in your environm
 ```sh
 yarn build
-cd mui && yarn build
+cd mui
````
```suggestion
cd mui/dist
```

jeffvg commented:
> There is a subtle change for how the icon font repo is published now. Previously, the package.json for that was at the root of the repository and we published from the root and included the subfolder called iconfont (or icon-font, can't remember). So importing the package, you had to import the font file from the subfolder (import 'CSSFILE.css' from '@brightlayer-ui/icons/iconfont').
> 
> THAT IS NO LONGER THE CASE.
> 
> We moved the icon-font folder into a dedicated package folder under /packages and it gets published from that folder directly (no more subfolder). So the new usage is: import 'CSSFILE.css' from '@brightlayer-ui/icons'

I see ok, so as additional general comment the build kicked these files up for changes in syntax on my local...

<img width="1635" alt="image" src="https://user-images.githubusercontent.com/37150565/207437374-45d6fad3-e482-4fa0-b434-dbf02591c50d.png">

joebochill commented on packages/mui/README.md:
```diff
@@ -72,13 +72,12 @@ This will populate the /icons directory with all of the icon components as well
+yarn build
 npm version < patch | minor | major >
-npm publish
+npm publish:package
```
```suggestion
npm publish
```

publish:package is an npm script that we created. `npm publish` is an NPM CLI command.
It either needs to be `yarn publish:package` or `npm publish`

joebochill commented on packages/mui/README.md:
````diff
@@ -72,13 +72,12 @@ This will populate the /icons directory with all of the icon components as well
 ```
-yarn build:full
-cd icons
+yarn build
````
```suggestion
yarn build
cd dist
```

joebochill commented on packages/svg/README.md:
````diff
@@ -15,7 +16,17 @@ or
 ```
 
-### Angular
+To install the Brightlayer UI icon font from NPM as a dependency for your project, you can run one of the following commands in your project root:
+
+```
+npm install --save @brightlayer-ui/icons
+or
+yarn add @brightlayer-ui/icons
+```
+
+You then need to reference the font in your application so that it is available for use.
````
Shouldn't be talking about icon font in the SVG readme

joebochill commented on packages/svg/README.md:
````diff
@@ -104,7 +113,20 @@ To change the color of the icon you are using, simply set the CSS color property
 ```
 
-### React
+### Icon Font Angular Usage
+
+In order to use the icon font, you will need to modify your angular.json file "styles" entries (there is one under "build" and one under "test") to include the icon font reference:
+
+```
+"styles": [
+    "src/styles.scss",
+    "./node_modules/@brightlayer-ui/icons/iconfont/BrightlayerUIIcons.css"
+],
+```
+
````
Do not talk about icon font in the SVG readme

joebochill commented on packages/svg/README.md:
````diff
@@ -116,7 +138,31 @@ import icon from '@brightlayer-ui/icons-svg/ICON_NAME.svg';
 > NOTE: If you will be using many of these icons in your application, we recommend you use [@brightlayer-ui/icons-mui](https://www.npmjs.com/package/@brightlayer-ui/icons-mui) or our [icon font](https://www.npmjs.com/package/@brightlayer-ui/icons) to simplify usage. This library makes more sense if you just need one or two icons or if you want to reduce the size of your bundle.
 
-### React-Native
+### Icon Font React Usage
+
+For React, you'll need to require the font in your top-level JS file (typically index.js).
+
+```
+require('@brightlayer-ui/icons/iconfont/BrightlayerUIIcons.css');
+```
+
+## Using the icon font in your application
+
+This will make the Brightlayer UI icons available to your code by using the appropriate class name (className for React) on a `<i>` element, e.g.:
+
+```
+<i class="blui-ICONNAME"></i>
+```
````
Do not talk about icon font in the SVG readme

joebochill commented on packages/svg/README.md:
````diff
@@ -116,7 +138,31 @@ import icon from '@brightlayer-ui/icons-svg/ICON_NAME.svg';
+<i class="blui-ICONNAME"></i>
+```
+
+### Applying Different Colors
+
+To change the color of the icon you are using, simply set the CSS color property. This property is inheritable, so if the property is not set on your icon, it will be inherited from the parent container.
+
+```
+<i class="blui-ICONNAME" style="color: red"></i>
+```
+
````
This is icon-font specific and shouldn't be in the svg readme

joebochill commented on packages/svg/README.md:
```diff
@@ -126,6 +172,27 @@ import Leaf from "@brightlayer-ui/icons-svg/leaf.svg";
+
+The icon font is a great way to include lots of icons in your application. If you only need a few, you can also check out these alternative packages:
+
+-   [SVG Icons](https://www.npmjs.com/package/@brightlayer-ui/icons-svg)
```
We are in the SVG repository. This should be a line item for the icon font.

joebochill commented:
> I see ok, so as additional general comment the build kicked these files up for changes in syntax on my local...

I think that's just an issue with prettier running vs. not running (we probably should put generated files in the prettierigore, but 🤷)

bkarambe commented:
> > There is a subtle change for how the icon font repo is published now. Previously, the package.json for that was at the root of the repository and we published from the root and included the subfolder called iconfont (or icon-font, can't remember). So importing the package, you had to import the font file from the subfolder (import 'CSSFILE.css' from '@brightlayer-ui/icons/iconfont').
> > THAT IS NO LONGER THE CASE.
> > We moved the icon-font folder into a dedicated package folder under /packages and it gets published from that folder directly (no more subfolder). So the new usage is: import 'CSSFILE.css' from '@brightlayer-ui/icons'
> 
> I see ok, so as additional general comment the build kicked these files up for changes in syntax on my local...
> 
> <img alt="image" width="1635" src="https://user-images.githubusercontent.com/37150565/207437374-45d6fad3-e482-4fa0-b434-dbf02591c50d.png">

@jeffvg Sorry for that instruction in the description. I have updated the same now. There is no need of manual copying of node modules. There will not be icon-font folder in node modules after yarn build.

bkarambe commented:
> > I see ok, so as additional general comment the build kicked these files up for changes in syntax on my local...
> 
> I think that's just an issue with prettier running vs. not running (we probably should put generated files in the prettierigore, but 🤷)

I will add those to prettier ignore

jeffvg commented:
@josephpboyle you mentioned earlier about - 
@bkarambe 

- /packages/rn-vector/Font and /packages/rn-vector/GlyphMaps folders with a font a a json file in them, respectively (THIS PART OF THE BUILD IS BROKEN AND NEEDS TO BE FIXED...it's trying to copy files from the wrong location. We also need to create those files if they don't exist or it will copy the file to a file instead of a folder)

Should the rn-vector/Font have a json file? Currently I have this after yarn && yarn build

<img width="243" alt="image" src="https://user-images.githubusercontent.com/37150565/207710530-975e55ad-19b7-4603-95d5-8b2117040ae3.png">

joebochill commented:
> @josephpboyle you mentioned earlier about - @bkarambe
> 
> * /packages/rn-vector/Font and /packages/rn-vector/GlyphMaps folders with a font a a json file in them, respectively (THIS PART OF THE BUILD IS BROKEN AND NEEDS TO BE FIXED...it's trying to copy files from the wrong location. We also need to create those files if they don't exist or it will copy the file to a file instead of a folder)
> 
> Should the rn-vector/Font have a json file? Currently I have this after yarn && yarn build

This is correct, the .ttf file is in the Fonts folder and the .json is in the GlyphMaps

joebochill commented on packages/icon-font/CHANGELOG.md:
````diff
@@ -1,5 +1,15 @@
+-   Changed import path of icon font files to `icons`.
+
+```sh
+import '@brightlayer-ui/icons/BrightlayerUIIcons.css'
````
Show a before/after code snippet here using the 'diff' language and show it for both react and angular.

e.g.,

```diff
- import '@brightlayer-ui/icons/iconfont/BrightlayerUIIcons.css'
+ import '@brightlayer-ui/icons/BrightlayerUIIcons.css'
```

joebochill commented on packages/icon-font/package.json:
```diff
@@ -14,7 +14,9 @@
-        "prettier:check": "prettier \"*.md\" --check"
+        "prettier:check": "prettier \"*.md\" --check",
+        "publish:package": "set npm_config_yes=true && npx -p @brightlayer-ui/publish blui-publish",
+        "tag:package": "set npm_config_yes=true && npx -p @brightlayer-ui/tag blui-tag"
```
I believe the 'tag' command should also include an '-s blui-icons-svg' flag to add a suffix. Correct me if I'm wrong @emclaug2 .

joebochill commented on packages/rn-vector/package.json:
```diff
@@ -63,5 +63,11 @@
-    }
+    },
+    "dependencies": {
+        "@brightlayer-ui/prettier-config": "^1.0.3",
+        "eslint-config-prettier": "^8.5.0",
+        "prettier": "^2.8.1"
```
these need to be in devDependencies only.

'dependencies' implies that we need to install these packages for our icons to work at runtime, but we only use these to format the code prior to building and publishing.

joebochill reviewed (commented):
Looking good, just a couple minor things I saw outstanding.

daileytj commented:
<img width="787" alt="Screen Shot 2022-12-19 at 9 05 59 AM" src="https://user-images.githubusercontent.com/13989985/208443819-78d268df-5531-4845-9933-79b4010ae1a8.png">
Update the imports for this so that the proper svgs render

daileytj reviewed (dismissed):
Approving, but since I did work on this I'm going to let someone else check this out and hit the green button.

bkarambe commented:
> Approving, but since I did work on this I'm going to let someone else check this out and hit the green button.

I think there was no need of importing svg icons individually. Let me add a commit

joebochill commented:
@bkarambe — @daileytj was right. You need to import the svg files individually. With the changes you made, you're not using the icons-svg package at all...you switched the demo the the MUI components, which we already have on the other page.

joebochill commented on packages/icon-font/package.json:
```diff
@@ -14,7 +14,9 @@
-        "prettier:check": "prettier \"*.md\" --check"
+        "prettier:check": "prettier \"*.md\" --check",
+        "publish:package": "set npm_config_yes=true && npx -p @brightlayer-ui/publish blui-publish",
+        "tag:package": "set npm_config_yes=true && npx -p @brightlayer-ui/tag blui-tag -s blui-icons-svg"
```
This is not the icons-svg package (its ust icons...aka icon font)...we should not be tagging it as icons-svg

joebochill commented on packages/mui/PUBLISHING.md:
````diff
@@ -14,7 +14,7 @@ If you need to publish a package manually, you can run the following commands fr
 ```sh
 yarn build
-cd mui && yarn build
+cd mui/dist
````
These instructions should be relative to the root folder of the package...so:
```
cd dist
```
instead of
```
cd mui/dist
```

joebochill commented:
Still haven't seen a PR on the doc-it side to address the new urls/paths...when are you planning to do that?

bkarambe commented:
> Still haven't seen a PR on the doc-it side to address the new urls/paths...when are you planning to do that?

Will raise PR soon

joebochill commented:
Looks much better!
